src/app/data.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { startWith, shareReplay } from 'rxjs/operators';
+
+import { InputData } from './inputData';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DataService {
+  inputData: InputData = { input: "", inputFormat: "", outputFormat: "", additionalOptions: "" };
+  output: BehaviorSubject<String>= new BehaviorSubject<String>("");
+
+  constructor() {}
+
+  submit(): void {
```

Comment:
Once this is moved into it's separate PR, please update the submit route to use the correct logic we discussed.

example:

def submit(input_string):ObservableString>{
     return this.http.patch<string>(YOUR_URL, input_string);
}